app/repositories/application/GeneralApplicationRepository.scala

```diff
@@ -364,15 +365,30 @@ class GeneralApplicationMongoRepository(timeZoneService: TimeZoneService,
       BSONDocument("applicationStatus" -> BSONDocument("$ne" -> WITHDRAWN))
     )))
 
-  override def findFailedTestForNotification(failedTestType: FailedTestType): Future[Option[NotificationFailedTest]] = {
-    val query = BSONDocument("$and" -> BSONArray(
-      BSONDocument("applicationStatus" -> failedTestType.appStatus),
-      BSONDocument(s"progress-status.${failedTestType.notificationProgress}" -> BSONDocument("$ne" -> true)),
-      BSONDocument(s"progress-status.${failedTestType.receiveStatus}" -> true)
-    ))
+  override def findTestForNotification(notificationType: NotificationTestType): Future[Option[TestResultNotification]] = {
+    val query = Try{ notificationType match {
```

Comment:
Do we need this Try block?  There are only success or fail types defined so we should never fall into the unknown match.  I think we'll get an inexhaustive match compile error if this isn't updated if we add more types anyway.